### ISSUE TITLE
Incorrect warning of missing props validation

### ISSUE THREAD

**Issue by jakepusateri** [yannickcr/eslint-plugin-react#276]
I have a custom propTypes validator function, shown in the simplified component below
```javascript
import React from 'react';

var TestPropTypes = React.createClass({
    propTypes: {
        children: function (props, propName) {
            var allPanelChildren = props[propName].every(function (child) {
                return /Panel/.test(child.type.displayName);
            });
            if (!allPanelChildren) {
                return new Error('All children must be of type Panel.');
            }
        }
    },
    render: function () {
        return this.props.children;
    }
});
export default TestPropTypes;
```

When I run eslint the following warning is listed:

    6:42  warning  'every' is missing in props validation for TestPropTypes  react/prop-types

However, the only instance of 'every' in the file is not accessing a prop, but is instead the array method every.

eslint-plugin-react version: 3.6.3

**Comment by yannickcr** [comment]
Working on it.

**Comment by yannickcr** [comment]
Should be fixed by 2749a0b15bdc7747fe30cdebfde7f54cd41ba75e in `v3.7.0`. Let me know if it's not the case.

**Comment by larrybotha** [comment]
@yannickcr getting this on eslint-plugin-react v3.16.1

**Comment by ljharb** [comment]
@larrybotha can you provide the code that's being warned on, and can you try it in the latest version of the plugin (v4.3.0)?

**Comment by larrybotha** [comment]
This is bizarre... but it doesn't happen anymore... perhaps after fixing other linting errors. If that's even possible.

It was a simple function on an array prop:
```js
hasData() {
  return this.props.fields.every(field => !!field);
},
```

Anyways, with v.4.3.0 it's all good too 👍
